services/common/src/components/permits/ReportPermitRequirementForm.tsx

```diff
@@ -21,108 +19,239 @@ import RenderRadioButtons from "@mds/common/components/forms/RenderRadioButtons"
 import { FORM } from "@mds/common/constants/forms";
 import { MINE_REPORT_SUBMISSION_CODES, REPORT_TYPE_CODES } from "@mds/common/constants/enums";
 import { REPORT_FREQUENCY_HASH, REPORT_MINISTRY_RECIPIENT_HASH, REPORT_REGULATORY_AUTHORITY_CODES_HASH } from "@mds/common/constants/strings";
-import LinkButton from "@mds/common/components/common/LinkButton";
 import {
+  createMineReportPermitRequirement,
   deleteMineReportPermitRequirement,
   updateMineReportPermitRequirement,
 } from "@mds/common/redux/slices/mineReportPermitRequirementSlice";
 import { deleteConfirmWrapper } from "@mds/common/components/common/ActionMenu";
 import { usePermitConditions } from "@mds/common/components/permits/PermitConditionsContext";
+import { getMineReportPermitRequirementsByAmendment, getPermitConditionCategories } from "@mds/common/redux/selectors/permitSelectors";
+import CoreButton from "../common/CoreButton";
+import RenderTreeSelect, { parseTreeValues } from "../forms/RenderTreeSelect";
+import RenderSubmitButton from "../forms/RenderSubmitButton";
+import { closeModal } from "@mds/common/redux/actions/modalActions";
 
 
 interface ReportPermitRequirementProps {
-  onSubmit?: (values: Partial<IMineReport>) => void | Promise<void>;
-  permitGuid: string;
-  condition: IPermitCondition;
-  modalView?: boolean;
+  condition?: IPermitCondition;
+  isModal?: boolean;
   mineReportPermitRequirement?: IMineReportPermitRequirement;
   canEditPermitConditions: boolean;
   refreshData: () => Promise<void>;
-  currentAmendment: IPermitAmendment;
-  mineGuid: string;
 }
 
 export const ReportPermitRequirementForm: FC<ReportPermitRequirementProps> = ({
-  onSubmit,
   condition,
-  modalView = true,
+  isModal = false,
   mineReportPermitRequirement,
   canEditPermitConditions,
   refreshData,
-  permitGuid,
-  currentAmendment,
-  mineGuid,
 }) => {
-  const { loading } = usePermitConditions();
+  const formName = `${FORM.ADD_REPORT_TO_PERMIT_CONDITION}-${condition?.permit_condition_id ?? mineReportPermitRequirement?.mine_report_permit_requirement_id}`;
```

Comment:
this diff doesn't render by default because it's too large but a lot of important changes happen here in this file.